lua/acf/scanner/scanner_sh.lua

```diff
@@ -0,0 +1,1472 @@
+            gui.EnableScreenClicker(false)
+            screenClickerEnabledBecauseOfUs = false
+        end
+    end)
+
+    hook.Add("CreateMove", "ACF_Scanner_BlockInputs", function(cmd)
+        if scanning.IsScannerActive() then
+            if lastAng == nil then
+                lastAng = cmd:GetViewAngles()
+            end
+
+            local delta = cmd:GetViewAngles() - lastAng
+            angOffset = angOffset + delta
+            angOffset.pitch = math.Clamp(angOffset.pitch, -90, 90)
+            cmd:SetViewAngles(lastAng)
+            lastAng = cmd:GetViewAngles()
+
+            local wheel = cmd:GetMouseWheel()
+            if wheel ~= 0 then
+                scanning.AddMouseZoom(50 * wheel)
+            end
+        else
+            lastAng = nil
+        end
+    end)
+
+    local wireframe = Material("models/wireframe")
+    local colorMat = Material("models/debug/debugwhite")
+    local physMeshCache = {}
+
+    local function drawEntityNoOutline(ent, color)
+        if not IsValid(ent) then return end
+
+        render.MaterialOverride(colorMat)
+        render.SetColorModulation(color.r / 255, color.g / 255, color.b / 255)
+        render.SetBlend(0.15)
+        render.DepthRange(0, 0.01)
+        render.SetLightingMode(1)
+        ent:DrawModel()
+
+        render.MaterialOverride()
+        render.SetColorModulation(1, 1, 1)
+        render.SetBlend(1)
+        render.DepthRange(0, 1)
+        render.SetLightingMode(0)
+    end
+
+    local function drawEntity(ent, color)
+        if not IsValid(ent) then return end
+
+        render.MaterialOverride(colorMat)
+        render.DepthRange(0, 0.0001)
+        render.SetLightingMode(2)
+
+        render.SetColorModulation(color.r / 255, color.g / 255, color.b / 255)
+        render.SetBlend(0.4)
+        --ent:DrawModel()
+
+        local m = Matrix()
+        m:Translate(Vector(0, 0, 200))
+        cam.PushModelMatrix(m)
+        ent:DrawModel()
+        cam.PopModelMatrix()
+
+        render.MaterialOverride()
+        render.SetColorModulation(1, 1, 1)
+        render.SetBlend(1)
+        render.DepthRange(0, 1)
+        render.SetLightingMode(0)
+    end
+
+    local function drawPhysMesh(ent, color)
+        if not IsValid(ent) then return end
+        local physobj = ent:GetPhysicsObject()
+        if not IsValid(physobj) then return end
+
+        if not physMeshCache[ent] or physMeshCache[ent]:expired() then
+            local mvs = physobj:GetMesh()
+            local rm = Mesh()
+            rm:BuildFromTriangles(mvs)
+
+            physMeshCache[ent] = {
+                birth = CurTime(),
+                mesh = rm,
+                expired = function(self)
+                    return (CurTime() - self.birth) > 2
+                end
+            }
+        end
+
+        local m = Matrix()
+        m:SetTranslation(ent:GetPos())
+        m:SetAngles(ent:GetAngles())
+        cam.PushModelMatrix(m)
+
+        render.SetStencilWriteMask(0xFF)
```

Comment:
Could you please document what these render operation blocks are doing for people looking at this in the future?